### PR TITLE
fix install scripts for v22+

### PR DESCRIPTION
**Pull request by skrdgraph**
@MichelDiz - please review this!

I have added an if-else block, to facilitate the checksum compare. This is an issue with the code here (& NOT our release scripts). The issue is as follows: 
- when we compare checksum SHA's, the script is `grep`'ing for the file-path which is wrong
- a SHA check should simply be a straightforward SHA check .. which is what the else-block is doing

For versions<22 -> IF section gets executed - retaining the older way
For versions>22 -> ELSE block gets executed - with the fix

Now that I have reviewed this code, I see a few issues in the way this is setup for users who want to play with the binary. We should talk about it separately - and clean this outside facing script.

-------

@Rajakavitha1 - how to test this script (from this PR): 

The original download document [here](https://dgraph.io/docs/deploy/download) tells the user to run this script using this command:
```
curl https://get.dgraph.io -sSf | bash
``` 

Because we have not rolled this out yet, the quick way to test this is by swapping the URL with like this: 
```
curl https://raw.githubusercontent.com/dgraph-io/Install-Dgraph/skrdgraph/fix_install_scripts_mess/getdgraph.sh -sSf | bash
```

Please validate this ^ & leave a comment here. 

@skrdgraph I just validated and it works. Thank you for the quick turnaround
```
Do you agree to the terms of the Dgraph Community License? [Y/n] y
Latest release version is v22.0.2.
Downloading checksum file for v22.0.2 build.
######################################################################## 100.0%
Download complete.
Comparing checksums for dgraph binaries
Downloading https://github.com/dgraph-io/dgraph/releases/download/v22.0.2/dgraph-linux-amd64.tar.gz
######################################################################## 100.0%
Download complete.
Inflating binaries (password may be required).
Dgraph binaries v22.0.2 have been installed successfully in /usr/local/bin.
Please visit https://dgraph.io/docs/get-started for further instructions on usage.
```
-------

@MichelDiz - if everything looks okay, can you roll this out?

- - -
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/install-dgraph/33)

### CODE REVIEW

**Comment by skrdgraph** [comment]
Installation goes through fine with this change on my end: 

```
Do you agree to the terms of the Dgraph Community License? [Y/n] y
Latest release version is v22.0.2.
Downloading checksum file for v22.0.2 build.
######################################################################## 100.0%
Download complete.
Comparing checksums for dgraph binaries
Downloading https://github.com/dgraph-io/dgraph/releases/download/v22.0.2/dgraph-linux-amd64.tar.gz
######################################################################## 100.0%
Download complete.
Inflating binaries (password may be required).
Dgraph binaries v22.0.2 have been installed successfully in /usr/local/bin.
Please visit https://dgraph.io/docs/get-started for further instructions on usage.
```

I really don't like this model. We should have an internal discussion on this. I want to understand the following: 
- how is this being used?
- target users?
- discord issue reports?
- how can we improve this?
- or do we drop this?

I think this is an important script, I just feel like we can do a better job in maintaining it. I look forward to a discussion here.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /tmp
 .DS_store
+.idea/*
```
Comment (mangalaman93) [inline .gitignore:3]: add a new line

File: getdgraph.sh [hunk 2]
```diff
@@ -173,7 +173,14 @@ printf "%b" "$RESET"
 		exit 1;
 	fi
 
-	dgraph=$(grep -m 1 "$install_path"/dgraph  /tmp/"$checksum_file" | grep -E -o '[a-zA-Z0-9]{64}')
+	if grep -Fxq "$install_path"/dgraph /tmp/"$checksum_file"
+	then
+		# for versions < 22, old team was hardcoding install paths in SHA file
```
Comment (mangalaman93) [inline getdgraph.sh:178]: let's just say here that we have modified our release process and an example of the structure would be useful.

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: getdgraph.sh (hunk 2 above)
Comment (skrdgraph) [reply getdgraph.sh:178]: +1 ^

**Review by mangalaman93 — APPROVED** [review]

**Review by MichelDiz — APPROVED** [review]

**Comment by MichelDiz** [comment]
The changes are live.

image

**Comment by skrdgraph** [comment]
Thanks @MichelDiz ...

**Pull request merged**